miden-tx/src/testing/mock_chain/mod.rs

```diff
@@ -528,6 +505,30 @@ impl MockChain {
         ));
     }
 
+    /// Convert `auth_method` into its corresponding authentication [`AccountComponent`] and a
+    /// [`BasicAuthenticator`] or `None` when [`Auth::NoAuth`] is passed.
+    fn build_authentication_component(
+        auth_method: Auth,
+    ) -> Option<(AccountComponent, BasicAuthenticator<ChaCha20Rng>)> {
```

Comment:
This could probably be a method on the `Auth` struct, right? Something like:
```rust
impl Auth {
    pub fn build_component(&self) -> Option<(AccountComponent, BasicAuthenticator<ChaCha20Rng>)> {
        ...
    }
}
```